### PR TITLE
[Performance](point query)Opimize partition prune for point query

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 1]
```diff
@@ -674,8 +679,17 @@ private Collection<Long> partitionPrune(PartitionInfo partitionInfo,
...
-        if (partitionInfo.getType() == PartitionType.RANGE) {
+        if (partitionInfo.getType() == PartitionType.RANGE
+                    && isPointQuery() && partitionInfo.getPartitionColumns().size() == 1) {
+            // short circuit, a quick path to find partition
+            ColumnRange filterRange = columnNameToRange.get(partitionInfo.getPartitionColumns().get(0).getName());
+            LiteralExpr lowerBound = filterRange.getRangeSet().get().asRanges().stream()
+                    .findFirst().get().lowerEndpoint().getValue();
+            LiteralExpr upperBound = filterRange.getRangeSet().get().asRanges().stream()
+                    .findFirst().get().upperEndpoint().getValue();
+            cachedPartitionPruner.update(keyItemMap, lowerBound, upperBound);
```
[the remark this answers is not in the document]
Comment (xiaokang) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:690]: There is potential risk of concurrent write and read access for lowerBound/upperBound variable of PartitionPruneV2ForShortCircuitPlan.

File: fe/fe-core/src/main/java/org/apache/doris/planner/PartitionPruneV2ForShortCircuitPlan.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    PartitionPruneV2ForShortCircuitPlan() {
+        super();
+    }
+
+    public boolean update(Map<Long, PartitionItem> keyItemMap, LiteralExpr lowerBound, LiteralExpr upperBound) {
+        this.lowerBoundLiteral = lowerBound;
+        this.upperBoundLiteral = upperBound;
+        // interval to update partitionRangeMapByLiteral
+        long partitionRangeMapUpdateIntervalS = 10;
+        if (System.currentTimeMillis() - lastPartitionRangeMapUpdateTimestampS
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/planner/PartitionPruneV2ForShortCircuitPlan.java:52]: second

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 3]
```diff
@@ -199,6 +200,10 @@ public class OlapScanNode extends ScanNode {
 
     private boolean shouldColoScan = false;
 
+    // cached for prepared statement to quickly prune partition
+    private final PartitionPruneV2ForShortCircuitPlan cachedPartitionPruner =
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:204]: no only short circuit plan

File: fe/fe-core/src/main/java/org/apache/doris/planner/PartitionPruneV2ForShortCircuitPlan.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    > partitionRangeMapUpdateIntervalS * 1000) {
+            partitionRangeMapByLiteral = new RangeMap<>();
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/planner/PartitionPruneV2ForShortCircuitPlan.java:54]: it's single threaded

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:690]: done

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 5]
```diff
@@ -674,8 +679,17 @@ private Collection<Long> partitionPrune(PartitionInfo partitionInfo,
         } else {
             keyItemMap = partitionInfo.getIdToItem(false);
         }
-
-        if (partitionInfo.getType() == PartitionType.RANGE) {
+        if (partitionInfo.getType() == PartitionType.RANGE
+                    && isPointQuery() && partitionInfo.getPartitionColumns().size() == 1) {
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**